runtime/test/mock-slot-composer.js

```diff
@@ -53,21 +66,26 @@ class MockSlotComposer extends SlotComposer {
     }
   }
 
-  initializeRecipe(recipe) {
-    super.initializeRecipe(recipe);
-  }
-
+  /** @method newExpectations()
+   * Creates a new group of expectations (that may occur in arbitrary order.
+   */
   newExpectations() {
     this.expectQueue.push([]);
     return this;
   }
 
+  /** @method expectContentItemsNumber(num, content)
+   * Returns true, if the number of items in content's model is equal to the given number.
+   */
   expectContentItemsNumber(num, content) {
     assert(content.model.items, `Content model doesn\'t have items (${num} expected}`);
     assert(content.model.items.length <= num, `Too many items (${content.model.items.length}), while only ${num} were expected.`);
     return content.model.items.length == num;
   }
 
+  /** @method expectContentItemsNumber(num, content)
+   * Adds an optional rendering expectation - a group of expectations is considered satifsied if only optional expectation remain.
+   */
```

Comment:
Can the optional rendering expectation still be satisfied after entering into the next Expectations group?

Reply:
Only within its own group, added it to the comment too.
Also in parallel i added `skipOptional` method, where before verifying expectations mockSlotComposer withdraws all optional ones. Currently it works with setTimeout and is kind of hacky, but i'm working on a better solution.